jpegenc/subblocks/dct_1d/interfaces.py

```diff
@@ -0,0 +1,117 @@
+        self.y24 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y25 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y26 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y27 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y30 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y31 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y32 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y33 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y34 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y35 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y36 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y37 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y40 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y41 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y42 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y43 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y44 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y45 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y46 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y47 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y50 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y51 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y52 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y53 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y54 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y55 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y56 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y57 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y60 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y61 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y62 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y63 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y64 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y65 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y66 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y67 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y70 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y71 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y72 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y73 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y74 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y75 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y76 = Signal(intbv(0, min=-out_range, max=out_range))
+        self.y77 = Signal(intbv(0, min=-out_range, max=out_range))
```

Comment:
First, why not use a list-of-signals?

If an LoS can't be used, there is a little python trick you can do to build the discrete outputs.

```
class outputs(object):
    def __init__(self):
        self._build_output_attributes()
    def _build_output_attributes(self):
        # 80 should be an argument (modular / scalable)
        for ii in range(80):
            self.__dict__['y{:02d}'.format(ii)] = Signal(intbv(0, min=-8, max=8))

outs = outputs()
pprint(vars(outs))

{'y00': Signal(intbv(0)),
 'y01': Signal(intbv(0)),
 'y02': Signal(intbv(0)),
 'y03': Signal(intbv(0)),
 'y04': Signal(intbv(0)),
 'y05': Signal(intbv(0)),
 'y06': Signal(intbv(0)),
 'y07': Signal(intbv(0)),
 'y08': Signal(intbv(0)),
 'y09': Signal(intbv(0)),
 'y10': Signal(intbv(0)),
 'y11': Signal(intbv(0))
...
```